### PR TITLE
new fx

### PR DESCRIPTION
**Pull request by eisascience**
Classification FXs might need minor debug

### CODE REVIEW

File: seurat/CellClassificationFXs.R [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+
+    return(list(yhat.DF = testing.data.yhat,
+                classification.levels = levels(testing.data.yhat[,1]),
+                Available.Classifiers = Available.Classifiers,
+                log10T = log10T,
+                DGEcolNames = colnames(testing.data),
+                DGErowNames = rownames(testing.data),
+                MultiClassifierResults = ifelse(returnTraining, MultiClassifierResults, "NULL")))
+
+  } else {
```
Comment (bbimber) [inline seurat/CellClassificationFXs.R]: Why not place this check first, and abort if it fails?  this saves having to wrap the entire first part of the method in an if()?

File: seurat/CellClassificationFXs.R [hunk 2]
```diff
@@ -0,0 +1,605 @@
...
+  if(!returnScore){
+
+    for (colName in colnames(genes.scores.use)) {
+      object[[colName]] <- genes.scores.use[colnames(object),colName]
+    }
+
+    gc(verbose = FALSE)
+    return(object)
+
+  } else {
```
Comment (bbimber) [inline seurat/CellClassificationFXs.R:168]: Same as my comment below.  in general, if you have an if/else, and one of these conditions is a few lines of code that stops/exits the method (like the else here), it tends to be cleaner to put that upfront, and then you dont need to wrap the longer if() portion

File: seurat/CellClassificationFXs.R [hunk 3]
```diff
@@ -0,0 +1,605 @@
...
+
+    if(!is.na(save.path)){
+
+      save.path <- paste(gsub(".csv", "", save.path), nameID, "_DimReduxComps.csv", sep="")
+      write.csv(tempDT, file = save.path, row.names=TRUE, col.names = T)
+
+    }
+
+
+  } else {
```
Comment (bbimber) [inline seurat/CellClassificationFXs.R]: i'm commenting out of linear order, but same as the ones below.  might as well put the sanity checking above and just return if it fails.

**Review by bbimber — CHANGES_REQUESTED** [review]

**Comment by eisascience** [comment]
fixed sanity check style

**Pull request merged**